editoast/src/core/simulation.rs

```diff
@@ -66,6 +66,47 @@ pub struct PhysicsRollingStock {
     pub raise_pantograph_time: Option<u64>,
 }
 
+pub struct SimulationParameters {
+    pub tonnage: Option<f64>,
+    pub total_length: Option<f64>,
+}
+
+impl PhysicsRollingStock {
+    /// Create a new PhysicsRollingStock from a RollingStockModel and SimulationParameters
+    pub fn new(traction_engine: RollingStockModel, params: SimulationParameters) -> Self {
+        let traction_engine_length = traction_engine.length * 1000.0; // RollingStockModel length is in km, convert to m
+        let length = params
+            .total_length // Simulation params is in m
+            .unwrap_or(traction_engine_length)
+            .round() as u64;
+
+        let traction_engine_mass = traction_engine.mass; // RollingStockModel mass is in kg
+        let params_mass = params.tonnage.map(|m| m * 1000.0); // Simulation params is in tons, convert to kg
+        let mass = params_mass.unwrap_or(traction_engine_mass).round() as u64;
+
+        Self {
+            effort_curves: traction_engine.effort_curves,
+            base_power_class: traction_engine.base_power_class,
+            length,
+            mass,
+            max_speed: traction_engine.max_speed,
+            startup_time: (traction_engine.startup_time * 1000.0).round() as u64,
+            startup_acceleration: traction_engine.startup_acceleration,
+            comfort_acceleration: traction_engine.comfort_acceleration,
+            gamma: traction_engine.gamma,
+            inertia_coefficient: traction_engine.inertia_coefficient,
+            rolling_resistance: traction_engine.rolling_resistance,
+            power_restrictions: traction_engine.power_restrictions.into_iter().collect(),
+            electrical_power_startup_time: traction_engine
+                .electrical_power_startup_time
+                .map(|v| (v * 1000.0).round() as u64),
+            raise_pantograph_time: traction_engine
+                .raise_pantograph_time
+                .map(|v| (v * 1000.0).round() as u64),
+        }
+    }
+}
+
 impl From<RollingStockModel> for PhysicsRollingStock {
```

Comment:
This implementation should be removed since I don't believe it's used anymore. If it is, better call `::new(value, Default::default())` instead.